shared/src/main/java/com/couchbase/lite/internal/support/Log.java

```diff
@@ -385,49 +342,72 @@ public static void e(String tag, String formatString, Throwable tr, Object... ar
      * @param args         Variable number of Object args to be used as params to formatString.
      */
     public static void e(String tag, String formatString, Object... args) {
-        if (logger != null && isLoggingEnabled(tag, ERROR)) {
-            try {
-                logger.e(tag, String.format(Locale.ENGLISH, formatString, args));
-            } catch (Exception e) {
-                logger.e(tag, String.format(Locale.ENGLISH, "Unable to format log: %s", formatString), e);
-            }
+        String msg;
+        try {
+            msg = String.format(Locale.ENGLISH, formatString, args);
+        } catch (Exception e) {
+            msg = String.format(Locale.ENGLISH, "Unable to format log: %s (%s)", formatString, e.toString());
         }
+        sendToLoggers(LogLevel.ERROR, tag, msg);
     }
 
     public static void setLogLevel(LogDomain domain, LogLevel level) {
+        int actualLevel = level.equals(LogLevel.NONE) ? Log.NONE : Log.DEBUG;
         switch (domain) {
             case ALL:
-                enableLogging(DATABASE, level.getValue());
-                enableLogging(QUERY, level.getValue());
-                enableLogging(SYNC, level.getValue());
-                enableLogging(WEB_SOCKET, level.getValue());
-                enableLogging(C4LogDomain.BLIP, level.getValue());
-                enableLogging(C4LogDomain.SyncBusy, level.getValue());
+                enableLogging(DATABASE, actualLevel);
+                enableLogging(QUERY, actualLevel);
+                enableLogging(SYNC, actualLevel);
+                enableLogging(WEB_SOCKET, actualLevel);
+                enableLogging(C4LogDomain.BLIP, actualLevel);
+                enableLogging(C4LogDomain.SyncBusy, actualLevel);
                 break;
             case DATABASE:
-                enableLogging(DATABASE, level.getValue());
+                enableLogging(DATABASE, actualLevel);
                 break;
 
             case QUERY:
-                enableLogging(QUERY, level.getValue());
+                enableLogging(QUERY, actualLevel);
                 break;
 
             case REPLICATOR:
-                enableLogging(SYNC, level.getValue());
-                enableLogging(C4LogDomain.SyncBusy, level.getValue());
+                enableLogging(SYNC, actualLevel);
+                enableLogging(C4LogDomain.SyncBusy, actualLevel);
                 break;
 
             case NETWORK:
-                enableLogging(C4LogDomain.BLIP, level.getValue());
-                enableLogging(WEB_SOCKET, level.getValue());
+                enableLogging(C4LogDomain.BLIP, actualLevel);
+                enableLogging(WEB_SOCKET, actualLevel);
                 break;
         }
     }
 
     public static void enableLogging(String tag, int logLevel) {
```

Comment:
Do you know when this method is used? It's public as well.

Reply:
I don't think I added this.  It's used for LiteCore log levels.

Reply:
OK.